applications/zpc/components/zwave_command_classes/src/zwave_command_class_supervision.c

```diff
@@ -349,6 +353,36 @@ void zwave_command_class_supervision_on_send_data_complete(
   memcpy(&ongoing_session->tx_info, tx_info, sizeof(zwapi_tx_report_t));
 }
 
+static void zwave_command_class_supervision_on_version_attribute_update(
+  attribute_store_node_t updated_node, attribute_store_change_t change)
+{
+  if (change == ATTRIBUTE_DELETED) {
+    return;
+  }
+
+  zwave_cc_version_t version = 0;
+  attribute_store_get_reported(updated_node, &version, sizeof(version));
+
+  if (version == 0) {
+    return;
+  }
+
+  attribute_store_node_t endpoint_node
+    = attribute_store_get_first_parent_with_type(updated_node,
+                                                 ATTRIBUTE_ENDPOINT_ID);
+
+  attribute_store_node_t supervision_version_node
```

Comment:
Here instead of add + set you could shortcut like this:

```suggestion
attribute_store_set_child_reported(endpoint_node,
                                                               ATTRIBUTE_COMMAND_CLASS_SUPERVISION_ENABLED,
                                                               &supervision_enabled_flag,
                                                               sizeof(supervision_enabled_flag));
```

Because adding here unconditionally may just get the system to add it on restarts. Just test to restart the ZPC and see if the  ATTRIBUTE_COMMAND_CLASS_SUPERVISION_ENABLED are not appearing in duplicates.

Also I would suggest to set that no matter the version of the Supervision attribute.

Also note that restart will always flip this setting back to 1, so if you do not have an active UAM map setting it it would re-activate itself on restart.